init.sh

```diff
@@ -45,23 +45,46 @@ if [ "$(uname)" = "Darwin" ] ; then
   function timeout() { perl -e 'alarm shift; exec @ARGV' "$@"; }
 fi
 
-# Download the remote resource to a local file of the same name, if the
-# remote resource is newer.  Works around connections that hang.  Takes a
-# single command-line argument, a URL.
+# Download the remote resource to a local file of the same name.
+# Takes a single command-line argument, a URL.
+# Skips the download if the remote resource is newer.
+# Works around connections that hang.
 download_url() {
-    echo "Downloading $@"
+    if [ "$#" -ne 1 ]; then
+        echo "Illegal number of arguments"
+    fi
+    URL=$1
+    echo "Downloading ${URL}"
     if [ "$(uname)" = "Darwin" ] ; then
-        wget -nv -N "$@"
+        wget -nv -N "URL" && echo "Downloaded URL"
     else
-	BASENAME=`basename ${@: -1}`
-	if [ -f $BASENAME ]; then
-	    ZBASENAME="-z $BASENAME"
-	else
-	    ZBASENAME=""
-	fi
-	timeout 300 curl -s -S -R -L -O $ZBASENAME "$@" || (echo "retrying curl $@" && rm -f "$BASENAME" && curl -R -L -O "$@")
+        BASENAME=$(basename ${URL: -1})
+        if [ -f $BASENAME ]; then
+            ZBASENAME="-z $BASENAME"
+        else
+            ZBASENAME=""
+        fi
+        (timeout 300 curl -s -S -R -L -O $ZBASENAME "URL" || (echo "retrying curl URL" && rm -f "$BASENAME" && curl -R -L -O "URL")) && echo "Downloaded URL"
+    fi
+}
+
+# Download the remote resource and unzip it.
+# Takes a single command-line argument, a URL.
+# Skips the download if the local file of the same name is newer.
+# Works around connections that hang and corrupted downloads.
+download_url_and_unzip() {
+    if [ "$#" -ne 1 ]; then
+        echo "Illegal number of arguments"
+    fi
+    URL=$1
+    BASENAME=$(basename ${URL: -1})
+    download_url "$URL"
+    if ! unzip -o "$BASENAME" > /dev/null ; then
+        echo "retrying download and unzip"
+        rm -rf "$BASENAME"
+        download_url "${URL}"
+        unzip -o "$BASENAME"
```

Comment:
We probably want the init script to fail if the second attempt isn't successful, but this can go into a dedicated change.

Reply:
I agree.  Failing the init script is a separate and desirable change.